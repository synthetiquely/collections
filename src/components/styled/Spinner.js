import React from 'react';
import glamorous from 'glamorous';
import { css } from 'glamor';

const animationStyles = () => {
  const spin = css.keyframes({
    '0%': { transform: 'rotate(0deg)' },
    '100%': { transform: 'rotate(360deg)' },
  });
  return {
    display: 'block',
    width: '60px',
    height: '60px',
    margin: '5px',
    border: '5px solid #ffdb4d',
    borderTopColor: 'transparent',
    borderRadius: '50%',
    animation: `${spin} 1s linear infinite`,
  };
};

const Overlay = glamorous.div({
  position: 'fixed',
  top: '0',
  right: '0',
  bottom: '0',
  left: '0',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  backgroundColor: '#fff',
});

const Spinner = glamorous.span(animationStyles);

export default ({ overlayed, ...rest }) => {
  if (overlayed) {
    return (
      <Overlay>
        <Spinner {...rest} />
      </Overlay>
    );
  }
  return <Spinner {...rest} />;
};
