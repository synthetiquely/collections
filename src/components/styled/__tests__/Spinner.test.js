import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Spinner, { Overlay } from '../Spinner';

describe('Spinner Component', () => {
  it('should render and match its own snapshot', () => {
    const wrapper = mount(<Spinner />);
    const tree = renderer.create(<Spinner />).toJSON();

    expect(wrapper).toHaveLength(1);
    expect(tree).toMatchSnapshot();
  });

  it('should render Overlay component, if overlayed is true', () => {
    const wrapper = mount(<Spinner overlayed />);

    expect(wrapper.find(Overlay)).toHaveLength(1);
  });
});
