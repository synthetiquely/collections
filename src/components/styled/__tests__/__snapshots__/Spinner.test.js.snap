// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Component should render and match its own snapshot 1`] = `
<span
  className="css-1kv7myz"
/>
`;
